cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -116,10 +118,10 @@ function generate_prune_whitelist_flags() {
 # besides the default ones.
 extra_prune_whitelist=
 if [ -n "${KUBECTL_EXTRA_PRUNE_WHITELIST:-}" ]; then
-  extra_prune_whitelist=( ${KUBECTL_EXTRA_PRUNE_WHITELIST:-} )
+  extra_prune_whitelist=( "${KUBECTL_EXTRA_PRUNE_WHITELIST:-}" )
 fi
-prune_whitelist=( ${KUBECTL_PRUNE_WHITELIST[@]}  ${extra_prune_whitelist[@]} )
-prune_whitelist_flags=$(generate_prune_whitelist_flags ${prune_whitelist[@]})
+prune_whitelist=( "${KUBECTL_PRUNE_WHITELIST[@]}"  "${extra_prune_whitelist[@]}" )
```

Comment:
These are also wrong - they will convert the whitelists into single string variables

Reply:
I think the problem is that `prune_whitelist_flags` is busted and you are working around it.  Let;s fix that function, document the inputs and outputs, and then fix callers.